### ISSUE TITLE
Implement adaptive learning rate controller

### ISSUE THREAD

**Issue by abuzarmahmood** [abuzarmahmood/blechRNN#6]

**Comment by abuzarmahmood** [comment]
The suggestions and changes made by the agents to implement an adaptive learning rate controller for fitting RNNs in the `abuzarmahmood/blechRNN` repository are as follows:

1. **Files to Modify:**
   - **`model.py`:** This file likely contains the implementation of the recurrent neural network model. Modifications might involve integrating an adaptive learning rate mechanism within the model's training loop or optimizer settings.
   - **`run_model.py`:** This script is used to run the model on data and generate outputs. Changes might be necessary to accommodate any new parameters or configurations related to the adaptive learning rate.
   - **`train.py`:** This file handles model training and prediction. It is the most likely candidate for implementing the adaptive learning rate controller, as it deals directly with the training process.

2. **Specific Changes in `train.py`:**
   - **Import the Learning Rate Scheduler:**
     ```python
     from torch.optim.lr_scheduler import ReduceLROnPlateau
     ```
   - **Initialize the Scheduler:**
     ```python
     scheduler = ReduceLROnPlateau(optimizer, mode='min', factor=0.1, patience=10, verbose=True)
     ```
   - **Update the Scheduler:**
     ```python
     if cross_val_bool and (i % 100 == 99):
         test_out, _ = net(test_inputs)
         test_out = test_out.reshape(-1, output_size)
         test_labels = test_labels.reshape(-1, output_size)
         test_loss = criterion(test_out, test_labels)
         cross_val_loss[i] = test_loss.item()
         scheduler.step(test_loss)  # Update the scheduler
         cross_str = f'Cross Val Loss: {test_loss.item():0.4f}'
     ```

3. **Update `run_model.py` if Necessary:**
   - Ensure any specific configurations or parameters related to the adaptive learning rate are included in the call to `train_model`.

These changes will implement an adaptive learning rate controller using PyTorch's `ReduceLROnPlateau` scheduler, which reduces the learning rate when a metric has stopped improving, aiding in better convergence during training.

TERMINATE

---
*This response was automatically generated by blech_bot*
